packages/@o3r/extractors/src/core/comparator/metadata-comparison.helper.ts

```diff
@@ -0,0 +1,135 @@
+import type { BuilderContext, BuilderOutput } from '@angular-devkit/architect';
+import { getPackageManagerInfo, type PackageManagerOptions, type SupportedPackageManagers, type WorkspaceSchema } from '@o3r/schematics';
+import globby from 'globby';
+import { existsSync, readFileSync } from 'node:fs';
+import { join, posix } from 'node:path';
+import type { MetadataComparator, MigrationData, MigrationMetadataCheckBuilderOptions } from './metadata-comparator.interface';
+import { getFilesFromRegistry, getLatestMigrationMetadataFile, getLocalMetadataFile, getVersionRangeFromLatestVersion } from './metadata-files.helper';
+import type { JsonObject } from '@angular-devkit/core';
+
+// TODO : create a bespoke type for the errors, or do we accept them to be native JS Errors?
+function checkMetadataFile(lastMetadataFile: any, newMetadataFile: any, migrationData: MigrationData<any>[], isBreakingChangeAllowed: boolean, comparator: MetadataComparator<any>): Error[] {
+  const errors = [] as Error[];
+  const newMetadataArray = comparator.getArray(newMetadataFile);
+  const lastMetadataArray = comparator.getArray(lastMetadataFile);
+  for (const lastValue of lastMetadataArray) {
+    const isInNewMetadata = newMetadataArray.some((newValue) => comparator.isSame(newValue, lastValue));
+    if (!isInNewMetadata) {
+      if (!isBreakingChangeAllowed) {
+        errors.push(new Error(`Property ${comparator.getName(lastValue)} is not present in the new metadata and breaking changes are not allowed`));
```

Comment:
Can't we provide more information about the breaking changes ?
maybe we can change the `isSame` method to have a method that already return an array of Error